File: ShortestRepetition/sr.rb
# shortest repetition soln in rb for codeeval by steven a dunn

File.open(ARGV[0], 'r').each_line do |line|
    line.chomp!
    period = 1
    subSeq = line[0]
    (1...line.length).each do |i|
        if line[period..-1].index(subSeq) != 0
            period += 1
            subSeq += line[i]
        end
    end
    puts period
end

File: ShortestRepetition/c/sr.c
// shortest repetition soln in c for code eval by steven a dunn

#include <stdio.h>
#include <string.h>

int main (int argc, char* const argv[])
{
    FILE *fp = fopen(argv[1], "r");
    if (fp != NULL)
    {
        char line[512];
        while (fgets(line, sizeof(line), fp))
        {
            for (int i = 0; i < sizeof(line); ++i)
                if (line[i] == '\n')
                    line[i] = '\0';

            char firstChar = line[0];
            int repIdx = 1;

            for(int i = 1; i < strlen(line); ++i)
            {
                if(line[i] == firstChar)
                    break;
                ++repIdx;
            }
            printf("%d\n", repIdx);
        }
        fclose(fp);
    }
    return 0;
}